sklearn/tests/test_isotonic.py

```diff
@@ -346,3 +346,31 @@ def test_isotonic_zero_weight_loop():
 
     # This will hang in failure case.
     regression.fit(x, y, sample_weight=w)
+
+
+def test_fast_predict():
+    # test that the faster prediction (https://github.com/scikit-learn/scikit-learn/pull/6206)
+    # change doesn't affect out-of-sample predictions.
+    rng = np.random.RandomState(123)
+    # np.random.seed(123)
+    N = 10**3
+    # X values over the -10,10 range
+    training_X = 20.0 * rng.rand(N) - 10
+    training_Y = np.less(
+        rng.rand(N),
+        1.0 / (1.0 + np.exp(-training_X))
+        ).astype('int64')
+    
+    slow_model = IsotonicRegression(y_min=0, y_max=1, out_of_bounds="clip")
+    fast_model = IsotonicRegression(y_min=0, y_max=1, out_of_bounds="clip")
+    
+    slow_model.fit(training_X, training_Y)
+    #rebuild with ALL input data, not just necessary
+    slow_model._build_f(slow_model.X_, slow_model.y_)
```

Comment:
This will raise a deprecation warning we we don't want to trigger in our tests.

Furthermore we want to have this test still work in the future once we delete the `X_` and `y_` attributes (in version 0.20). Instead please build `X_unique` and `y_isotonic` as local variables in the tests itself by calling `_make_unique` and `isotonic_regression` directly here.